date-time/src/commonMain/kotlin/com/splendo/kaluga/datetime/timer/Timer.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ Copyright 2021 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+package com.splendo.kaluga.datetime.timer
+
+import com.splendo.kaluga.base.utils.firstInstance
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.flow.transformWhile
+import kotlin.time.Duration
+
+/** A timer ticking a certain [duration]. */
+interface Timer {
+    /** Timer duration. */
+    val duration: Duration
+    /** Current state of the timer. */
+    val state: StateFlow<State>
+
+    sealed interface State {
+        sealed interface Running : State {
```

Comment:
Making this sealed makes it impossible to extend from outside of the module. Which makes it impossible to make a custom or mockable timer (especially since the TimerStateRepo is a private class).

Reply:
fixed

Reply:
You only need to remove the sealed from `State` and `NotRunning`.

This will create an exhaustable when for types `State` and `NotRunning` and will still allow mock states to be created for all the sub states.

Reply:
done